### PR TITLE
Feat/export

### PR DESCRIPTION
**Pull request by cloeser**
Adds document export functionality

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ here.
   and paste it in `CSAF_COOKIE_SECRET`.
 - Create a database in CouchDB with the name specified in `CSAF_COUCHDB_DBNAME`
 - restart compose
+- (required for exports) install [pandoc](https://pandoc.org/installing.html)
```
Comment (rainer-exxcellent) [inline README.md:28]: Add information which is the recommended version of pandoc and weasyprint

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java [hunk 2]
```diff
@@ -387,19 +393,51 @@ public ResponseEntity<JsonNode> readTemplate(
...
             @PathVariable
             @Parameter(in = ParameterIn.PATH, description = "The ID of the advisory to export.")
             String advisoryId,
             @RequestParam(required = false)
             @Parameter(description = "The format in which the document shall be exported.")
             ExportFormat format
     ) {
-
         // only for debugging, remove when implemented
         LOG.info("exportAdvisory to format: {} {}", sanitize(format), sanitize(advisoryId));
```
Comment (rainer-exxcellent) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/rest/AdvisoryController.java:405]: change log to debug and remove sanitized information

File: README.md (hunk 1 above)
Comment (rainer-exxcellent) [reply README.md:28]: add information how to configure the CSAF_COMPANY_LOGO_PATH

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -15,4 +15,5 @@ csaf.couchdb.password=${CSAF_COUCHDB_PASSWORD:}
 spring.security.oauth2.resourceserver.jwt.issuer-uri=${CSAF_KEYCLOAK_EXTERNAL_PROTOCOL:http}://${CSAF_KEYCLOAK_EXTERNAL_HOSTNAME:localhost}:${CSAF_KEYCLOAK_EXTERNAL_PORT:}/realms/${CSAF_REALM:}
 
 # templates
-csaf.document.templates.file=${CSAF_TEMPLATES_FILE:} 
\ No newline at end of file
+csaf.document.templates.file=${CSAF_TEMPLATES_FILE:}
+csaf.document.templates.companyLogoPath=${CSAF_COMPANY_LOGO_PATH:}
```
Comment (rainer-exxcellent) [inline src/main/resources/application.properties:19]: add information about the CSAF_COMPANY_LOGO_PATH also to the env.example file

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/service/PandocService.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * Service for converting between different document formats using the
+ * <a href="https://pandoc.org/">pandoc</a> cli tool.
+ * <p>
+ * Requires to have the pandoc command in the current system path installed
+ * (availability can be checked by using the {@link PandocService#isReady()} method).
+ */
+@Service
+public class PandocService {
```
[the remark this answers is not in the document]
Comment (wofl) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/service/PandocService.java:26]: I refactored the common parts into an abstract base class that can be used for all cli tool calls. Both the PandocService and the WeasyprintService inherit from this new service and only include the parts specific to the individual cli tools.

**Review by rainer-exxcellent — APPROVED** [review]

**Pull request merged**